### DISCUSSION TITLE
Loading model in spacy 3 which was trained with spacy-nightly

### DISCUSSION THREAD

**Discussion by jricheimer** [explosion/spaCy#7861] [Help: Other Questions] [answered]
I know that people have asked here how to load in spacy 3.x models which they've trained with spacy 2.x (spoiler: it's impossible). But I have a model that I've trained with spacy 3-alpha (specifically 3.0.0rc2 with spacy-nightly) before the official 3.0 release. And I get an error if I try to load those models using spacy 3.0.

In particular, spacy 3.0 complains about the config file:

```
ConfigValidationError: 

Config validation error

batch_size   field required

{'lang': 'en', 'pipeline': ['transformer', 'textcat'], 'tokenizer': {'@tokenizers': 'spacy.Tokenizer.v1'}, 'disabled': [], 'before_creation': None, 'after_creation': None, 'after_pipeline_creation': None}
```

Is there a known list of updates to the config file such that a model trained with spacy 3.x-alpha can be loaded with 3.x?

**Comment by adrianeboyd** [comment, marked as answer]
You can try `spacy init fill-config` to fill in missing values, but there's a good chance that the model will no longer perform as expected. Some of the internal model implementations changed across the rc releases, so I know that `parser` and `ner` models will no longer work correctly. `transformer` and `textcat` may be fine, but I'm not sure. If you test the model on your eval data and the results are the same as with 3.0.0rc2, then it should probably be okay to use. But in general we'd strongly recommend retraining.

**Reply by jricheimer** [reply]
This worked. Thank you
